tests/unit/test_utils.py

```diff
@@ -438,3 +440,123 @@ def test_FedoraDistGit(tmpdir):
+
+    def test_no_remote(cls, tmpdir):
+        subprocess.run('git init'.split(), check=True, cwd=tmpdir)
+        subprocess.run(
+            'git config --local user.email lzachar@redhat.com'.split(),
+            check=True,
+            cwd=tmpdir)
+        subprocess.run(
+            'git config --local user.name LZachar'.split(),
+            check=True,
+            cwd=tmpdir)
+        tmt.Tree.init(str(tmpdir), None, None)
+        tmpdir.join('main.fmf').write('test: echo')
+        subprocess.run('git add main.fmf'.split(), check=True, cwd=tmpdir)
+        subprocess.run(
+            'git commit -m initial_commit'.split(),
+            check=True,
+            cwd=tmpdir)
+
+        test = tmt.Tree(str(tmpdir)).tests()[0]
+        val, msg = validate_git_status(test)
+        assert not val
+        assert "Failed to get remote branch" in msg
+
+    @pytest.mark.parametrize("use_path",
+                             [False, True], ids=["without path", "with path"])
+    def test_local_changes(cls, tmpdir, use_path):
+        origin = tmpdir.join('origin')
+
+        if use_path:
+            fmfroot = origin.join('fmfroot')
+        else:
+            fmfroot = origin
+        tmt.Tree.init(str(fmfroot), None, None)
+        subprocess.run('git init'.split(), check=True, cwd=origin)
+        subprocess.run(
+            'git config --local user.email lzachar@redhat.com'.split(),
+            check=True,
+            cwd=origin)
+        subprocess.run(
+            'git config --local user.name LZachar'.split(),
+            check=True,
+            cwd=origin)
+        fmfroot.join('main.fmf').write('test: echo')
+        subprocess.run('git add -A'.split(), check=True, cwd=origin)
+        subprocess.run(
+            'git commit -m initial_commit'.split(),
+            check=True,
+            cwd=origin)
+
+        mine = tmpdir.join('mine')
+        subprocess.run(
+            f'git clone {origin} {mine}'.split(),
+            check=True,
+            cwd=tmpdir)
+        mine_fmf_root = mine
+        if use_path:
+            mine_fmf_root = mine.join('fmfroot')
+        mine_fmf_root.join('main.fmf').write('test: echo ahoy')
+
+        test = tmt.Tree(str(mine_fmf_root)).tests()[0]
+        validation_result = validate_git_status(test)
+
+        assert validation_result == (
+            False, "Not committed changes in " + ('fmfroot/' if use_path else '') + "main.fmf")
+
+    def test_not_pushed(cls, tmpdir):
+        origin = tmpdir.join('origin')
+        fmfroot = origin
+
+        tmt.Tree.init(str(fmfroot), None, None)
+        subprocess.run('git init'.split(), check=True, cwd=origin)
+        subprocess.run(
+            'git config --local user.email lzachar@redhat.com'.split(),
+            check=True,
+            cwd=origin)
+        subprocess.run(
+            'git config --local user.name LZachar'.split(),
+            check=True,
+            cwd=origin)
+        fmfroot.join('main.fmf').write('test: echo')
```

Comment:
What about creating a `setup()` method to cover these common preparation steps?

Reply:
yep, fixture was created.